core/src/main/java/nu/yona/server/goals/entities/Goal.java

```diff
@@ -29,6 +32,10 @@ public static GoalRepository getRepository()
 	private ActivityCategory activityCategory;
 
 	private ZonedDateTime creationTime;
+	private ZonedDateTime endTime;
+
+	@OneToOne(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+	private Goal previousGoal;
```

Comment:
Consider to rename to `previousVersion`. This property is only used when the goal had a previous version, not when you delete a goal and create a goal in the same category.